src/starkpil/commit_pols_starks.hpp

```diff
@@ -31,33 +32,28 @@ class CommitPolStarks
+    CommitPolStarks* a;
 
 private:
     void *_pAddress;
     uint64_t _degree;
+    uint64_t _nCommitedPols;
 
 public:
-    CompressorCommitPolsStarks(void *pAddress, uint64_t degree) : a{
-                                                                      CommitPolStarks((Goldilocks::Element *)((uint8_t *)pAddress + 0), degree, 0),
-                                                                      CommitPolStarks((Goldilocks::Element *)((uint8_t *)pAddress + 8), degree, 1),
-                                                                      CommitPolStarks((Goldilocks::Element *)((uint8_t *)pAddress + 16), degree, 2),
-                                                                      CommitPolStarks((Goldilocks::Element *)((uint8_t *)pAddress + 24), degree, 3),
-                                                                      CommitPolStarks((Goldilocks::Element *)((uint8_t *)pAddress + 32), degree, 4),
-                                                                      CommitPolStarks((Goldilocks::Element *)((uint8_t *)pAddress + 40), degree, 5),
-                                                                      CommitPolStarks((Goldilocks::Element *)((uint8_t *)pAddress + 48), degree, 6),
-                                                                      CommitPolStarks((Goldilocks::Element *)((uint8_t *)pAddress + 56), degree, 7),
-                                                                      CommitPolStarks((Goldilocks::Element *)((uint8_t *)pAddress + 64), degree, 8),
-                                                                      CommitPolStarks((Goldilocks::Element *)((uint8_t *)pAddress + 72), degree, 9),
-                                                                      CommitPolStarks((Goldilocks::Element *)((uint8_t *)pAddress + 80), degree, 10),
-                                                                      CommitPolStarks((Goldilocks::Element *)((uint8_t *)pAddress + 88), degree, 11)},
-                                                                  _pAddress(pAddress), _degree(degree){};
-
-    static uint64_t numPols(void) { return NUM_COMMIT_POLS; }
+    CompressorCommitPolsStarks(void *pAddress, uint64_t degree, uint64_t nCommitedPols) : _pAddress(pAddress), _degree(degree), _nCommitedPols(nCommitedPols){
+        a = new CommitPolStarks[nCommitedPols];
+
+        for (uint64_t i = 0; i < nCommitedPols; i++) {
+            a[i] = CommitPolStarks((Goldilocks::Element *)((uint8_t *)pAddress + sizeof(Goldilocks::Element)*i), degree, i, nCommitedPols);
+        }
+    };
+    
+    uint64_t numPols(void) { return _nCommitedPols; }
 
     void *address(void) { return _pAddress; }
     uint64_t degree(void) { return _degree; }
-    uint64_t size(void) { return _degree * NUM_COMMIT_POLS * sizeof(Goldilocks::Element); }
+    uint64_t size(void) { return _degree * _nCommitedPols * sizeof(Goldilocks::Element); }
+    ~CompressorCommitPolsStarks() {};
```

Comment:
Since now `a` is a pointer allocated with `new`, it's necessary to release the memory with `delete[] a;` on the destructor